### PR TITLE
Admission webhook: Undo prepare-shutdown calls if last-downscale fails to store

### PR DESCRIPTION
**Pull request by seizethedave**
Fix a snag found in #146 where if the "downscaled" annotation/configmap fails to persist, the scale operation is denied, but the pods are not informed via `DELETE` that they should no longer shutdown.

### CODE REVIEW

File: pkg/admission/prep_downscale_test.go [hunk 1]
```diff
@@ -58,8 +61,13 @@ func TestValidDownscaleWithAnotherInProgress(t *testing.T) {
 }
 
 func TestInvalidDownscale(t *testing.T) {
-	testPrepDownscaleWebhook(t, 5, 3, withStatusCode(http.StatusInternalServerError))
-	testPrepDownscaleWebhookWithZoneTracker(t, 5, 3, withStatusCode(http.StatusInternalServerError))
+	testPrepDownscaleWebhook(t, 5, 3, withStatusCode(http.StatusInternalServerError), expectDeletes())
```
Comment (pr00se) [inline pkg/admission/prep_downscale_test.go:64]: should these `expectDeny()` as well?

File: pkg/admission/prep_downscale_test.go [hunk 2]
```diff
@@ -275,53 +331,48 @@ func testPrepDownscaleWebhook(t *testing.T, oldReplicas, newReplicas int, option
...
-			expectDeletes: 64,
 			expectErr:     true,
 		},
 		"last post fails": {
 			numEndpoints:  64,
 			lastPostsFail: 1,
-			expectDeletes: 64,
 			expectErr:     true,
 		},
 		"delete failures should still call all deletes": {
```
Comment (pr00se) [inline pkg/admission/prep_downscale_test.go:366]: is this test now identical to the "all posts fail" one above?

**Review by pr00se — COMMENTED** [review]
A couple questions, but overall LGTM

**Comment by seizethedave** [comment]
Thanks for the read.

**Review by pr00se — APPROVED** [review]
LGTM, thanks!

**Pull request merged**